edx_solutions_api_integration/courses/views.py

```diff
@@ -2659,24 +2659,25 @@ def post(self, request):
         course_structures = CourseStructure.objects.filter(course_id__in=course_ids)
         response_data = []
         for course_structure in course_structures:
-            for course_id in course_ids:
-                if course_structure.course_id != course_id:
-                    continue
-
-                blocks = course_structure.structure.get('blocks', {}).copy()
-                for block in blocks.values():
-                    block['name'] = block.pop('display_name')
-                    block['category'] = block.pop('block_type')
-
-                course = [block for block_id, block in blocks.items() if block['category'] == 'course'][0]
-                self._update_blocks(course, blocks)
-                course_data = {
-                    "id": str(course_id),
-                    "content": course['children']
-                }
-                response_data.append(course_data)
+            blocks = course_structure.structure.get('blocks', {})
+            for block in blocks.values():
+                block['name'] = block.pop('display_name')
+                block['category'] = block.pop('block_type')
+
+            course = self._get_course(blocks)
```

Comment:
```
for block in blocks.values():
    block['name'] = block.pop('display_name')
    block['category'] = block.pop('block_type')
course = self._get_course(blocks)
```

This code block can be optimized to:

```
course = None
for block in blocks.values():
    block['name'] = block.pop('display_name')
   block['category'] = block.pop('block_type')
    if block['category'] == 'course' and course == None:
        course = block
```

@nasirhjafri what do you say?

Reply:
@ihtram sounds good, lemme do a quick PR.